kythe/go/extractors/config/runextractor/backup/backup.go

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2018 The Kythe Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Package backup is a simple library for backing up a config file and restoring
```

Comment:
I think this package might also benefit from a `Release` function, or something like it, to clean up a tempfile that isn't needed any longer.

In fact, now that I think about it some more, what do you think about making this a type that keeps the state?

https://play.golang.org/p/X5uv8XAriVM

That avoids the caller having to keep around the temp file path, and makes it easy to defer a cleanup that may or may not need to happen depending on success.

Reply:
Sure that looks pretty clean.